reimplement skull logic and misc cleanup

Berthalamew opened the pull request:
- fix original bug in the process with the string id set for the Whuppopotamus skull
- add the slot for Cow Bell (regardless if it doesn't do anything we should probably just have it for reference)
- move string id modules and rename e_global_string_ids to e_global_string_id
- cleanup header references in headers already modified from above changes

- benefit of this is that it allows us to easily add extra skulls in the future.
example: skull to remove the 5 second fall timer in campaign

Berthalamew committed:
reimplement skull logic and arrays

- fix original bug in the process with the string id set for the Whuppopotamus skull
- add the slot for Cow Bell even though it isn't used
- allows us to easily add extra skulls in the future

Berthalamew committed:
string_id moves

- rename string_ids.h to string_id.h
- rename HaloStrings.h to global_string_ids.h and move to tag_files folder
- rename e_global_string_ids to e_global_string_id

Berthalamew committed:
cleanup header references in h2mod and files that reference h2mod.h

- removed headers references that don't need to be referenced

Berthalamew committed:
revert sln visual studio version upgrade

it decided to do it automatically?

Berthalamew committed:
update spacing in cheats header

nukeulater commented on xlive/Blam/Engine/game/cheats.h:
```diff
@@ -1,37 +1,41 @@
 #pragma once
 
-#include "Blam/Engine/game/game_globals.h"
+#define SKULL_COUNT 16
```
noticed you want to follow the bungie naming scheme, thus this should be `k_skull_count`

Berthalamew replied:
renamed

nukeulater commented on xlive/Blam/Engine/game/cheats.cpp:
```diff
@@ -0,0 +1,98 @@
+void __cdecl ice_cream_flavor_stock(const e_skull_type skull)
+{
+	typedef void(__cdecl* hud_clear_messages_t)();
+	hud_clear_messages_t p_hud_clear_messages = Memory::GetAddress<hud_clear_messages_t>(0x22CE83, 0x206863);
```
`auto` because it's pointless to do it twice

Berthalamew replied:
changed

nukeulater commented on xlive/Blam/Engine/game/cheats.cpp:
```diff
@@ -0,0 +1,98 @@
+typedef void(__cdecl* ice_cream_activation_t)(unsigned int skull);
+ice_cream_activation_t p_ice_cream_activation;
+
+bool skull_enabled[SKULL_COUNT] = { false };
```
`skull_enabled[k_skull_count] = { };`

Berthalamew replied:
changed

nukeulater commented on xlive/H2MOD.cpp:
```diff
@@ -696,10 +690,10 @@ __declspec(naked) void calculate_model_lod_detour()
 fn_c000bd114_t p_fn_c000bd114;
 
-bool __cdecl fn_c000bd114_IsSkullEnabled(e_skulls skull_index)
+bool __cdecl fn_c000bd114_IsSkullEnabled(e_skull_type skull_index)
```
might as well rename this function while at it, e.g. `is_skull_enabled_hook()`

Berthalamew replied:
Nice catch, although for the wrong reason.
We shouldn't even have this function in the first place since it's a duplicate detour.
Removed the function

nukeulater commented on xlive/H2MOD/GUI/imgui_integration/AdvancedSettings.cpp:
```diff
@@ -1,21 +1,23 @@
-#include "Util\Hooks\Hook.h"
+#include "Blam/Engine/Networking/NetworkMessageTypeCollection.h"
+
+#include "H2MOD/Modules/CustomMenu/CustomMenu.h"
```
if you want open a pull to update the includes once and for all, but not for the vendor or 3rd-party libraries

Berthalamew replied:
I'll open a PR to update them once Kantanomo's json config PR is merged

Berthalamew committed:
add requested changes

- remove unnecessary hook
- rename SKULL_COUNT to k_skull_count
- other misc changes
